csrc/multi_tensor_adam.cu

```diff
@@ -169,3 +281,42 @@ void multi_tensor_adam_cuda(
   AT_CUDA_CHECK(cudaGetLastError());
 
 }
+
+void multi_tensor_adam_capturable_cuda(
+  int chunk_size,
+  at::Tensor noop_flag,
+  std::vector<std::vector<at::Tensor>> tensor_lists,
+  at::Tensor lr,
+  const float beta1,
+  const float beta2,
+  const float epsilon,
+  at::Tensor step,
+  const int mode,
+  const int bias_correction,
+  const float weight_decay,
+  at::Tensor inv_scale)
+{
+  using namespace at;
+
+  DISPATCH_DOUBLE_FLOAT_HALF_AND_BFLOAT(
+    tensor_lists[0][0].scalar_type(), 0, "adam",
+    multi_tensor_apply<4>(
+      BLOCK_SIZE,
+      chunk_size,
+      noop_flag,
+      tensor_lists,
+      AdamCapturableFunctor<scalar_t_0>(),
+      beta1,
+      beta2,
+      step.DATA_PTR<int>(),
+      bias_correction,
+      epsilon,
+      lr.DATA_PTR<float>(),
+      (adamMode_t) mode,
+      weight_decay,
+      inv_scale.data_ptr<float>()); )
```

Comment:
IMHO, it's okay to not use `DATA_PTR`

Reply:
The thing is `lr` and `step` are `at::Tensor` (see [here](https://github.com/NVIDIA/apex/blob/930bfb59b343b7dc2dccb0ee9fd29e41a54b9853/csrc/multi_tensor_adam.cu#L285-L297)). Will it still be correct if I will remove `DATA_PTR`?

Reply:
oh I meant to say it's safe to use `data_ptr` instead like `inv_scale` as per https://github.com/NVIDIA/apex/blob/75f401e088ef88e7c85a57ecf70fb232235f0334/csrc/compat.h#L5-L9 (where `VERSION_GE_1_3` means the available pytorch's version >= 1.3)

Reply:
Got it. I see I also used the lowercase version in line 317...
I modified everything to be lowercase as you suggested.